sdk/java-sdk-spring/src/main/scala/kalix/spring/boot/KalixConfiguration.scala

```diff
@@ -45,7 +46,9 @@ object KalixConfiguration {
 class KalixConfiguration(applicationContext: ApplicationContext) {
 
   @Bean
-  def config(): Config = ConfigFactory.load()
+  def config(): Config = {
+    KalixRunner.prepareConfig(ConfigFactory.load())
```

Comment:
I initially wanted to run this "preparation" inside the KalixRunner, but that doesn't play well with the integration tests. 

It's ok to load the dev-mode.conf, each time. For now, a package application won't have `dev-mode.conf` and the preparation will have zero effect. The problem is when running integration tests. Then we need to disable it and the ordering of loading matters.